meeg_preprocessing/utils.py

```diff
@@ -151,15 +153,28 @@ def setup_provenance(script, results_dir, config=None, use_agg=True):
     - sets log file for sterr output
     - writes log file with runtime information
     """
+    import inspect
+    if script is None:
+        script = inspect.stack()[1][1]
+        if script[0] == '<':
+            script = '__main__'
```

Comment:
I'm not sure to follow.

My main point is that you cannot just copy and paste code that contains ```__script__``` to ipython. This is annoying because when we build a script, we go back and forth to the editing. We can't afford to exit python and reload the entire data everytime.